src/java.base/share/classes/java/lang/StringBuilder.java

```diff
@@ -471,8 +471,11 @@ public StringBuilder repeat(CharSequence cs, int count) {
     @Override
     @IntrinsicCandidate
     public String toString() {
+        if (length() == 0) {
+            return "";
+        }
         // Create a copy, don't share the array
-        return new String(this);
+        return new String(this, null);
```

Comment:
Ok, this got me a bit confused, but I think this just inlines the call to this constructor:

```
    public String(StringBuilder builder) {
        this(builder, null);
    }
```

Reply:
Yes, I was mostly reaching for increased consistency with `StringBuffer` here.

Reply:
Good then, thanks.